server/mockAndelaApi/index.js

```diff
@@ -1,86 +1,32 @@
-  if (!placement) {
-    return helpers.badResponse(res, { error: 'Placement data not found' });
-  }
-
-  // Validating the placement object fields
-  const requiredFields = Object.keys(samplePlacement);
-  const givenFields = Object.keys(placement);
-  const missingFields = [];
-  requiredFields.forEach(filed => !givenFields.includes(filed) && missingFields.push(filed));
-  if (missingFields.length) {
-    return helpers.badResponse(res, { error: 'The placement object is missing some required fields', missingFields });
-  }
-
-  // Check if the placement don't already exist.
-  let placementAlreadyExist = false;
-  if (req.path === '/placement/onboarding') {
-    placementAlreadyExist = helpers.checkPlacementExist(placementDatabase, placement, 'onboarding');
-  }
-  if (req.path === '/placement/offboarding') {
-    placementAlreadyExist = helpers.checkPlacementExist(placementDatabase, placement, 'offboarding');
-  }
-  if (placementAlreadyExist) {
-    return helpers.badResponse(res, { error: 'This placement already exist. Validated with the ID' });
-  }
-  next();
-};
-
-router.post('/placement/onboarding', validatePlacementData, (req, res) => {
-  storePlacement(req.body.placement, 'onboarding');
-  return helpers.successResponse(res, { message: 'Onboarding placement created', values: req.body.placement }, 201);
-});
-
-router.post('/placement/offboarding', validatePlacementData, (req, res) => {
-  storePlacement(req.body.placement, 'offboarding');
-  return helpers.successResponse(res, { message: 'Offboarding placement created', values: req.body.placement }, 201);
-});
-
-
-router.get('/placements', (req, res) => { // eslint-disable-line consistent-return
-  const { query: { status } } = req;
-  if (!status || !possiblePlacementStatus.includes(status)) {
-    return helpers.badResponse(res, { error: 'status query is invalid' });
-  }
-  if (status === 'Placed - Awaiting Onboarding') {
-    return helpers.successResponse(res, { values: placementDatabase.onboarding });
-  }
-  if (status === 'External Engagements - Rolling Off') {
-    return helpers.successResponse(res, { values: placementDatabase.offboarding });
-  }
-});
-
-router.post('/reset', (req, res) => {
-  placementDatabase.offboarding = [];
-  placementDatabase.onboarding = [];
-  return helpers.successResponse(res, { message: 'Placement database successfully reset' });
-});
-
-router.all('*', (req, res) => {
-  proxy.web(req, res, { target: 'https://api-prod.andela.com/api/v1', changeOrigin: true });
+const { list } = slackClient.users;
+
+/**
+ * Generates random number of placements
+ *
+ * @param {String} status The status of the placement to generate
+ *
+ * @returns {Array} The list of placements generated
+ */
+async function generatePlacements(status) {
+  const placements = [];
+  for (let index = 0; index < faker.random.number(5); index++) {
+    const { members } = await list();
+    const emails = members.map(member => member.profile.email).filter(email => email);
```

Comment:
```suggestion
    const emails = members.map(member => member.profile.email).filter(email => email);
```
I am kind of curious why use filter after the map function already returns the emails

Reply:
This is because there are some `undefined` items after the mapping. Owing to the fact that some members of the slack workspace (slack bots) do not have an `email`

Reply:
Have replaced with `reduce`, more code/less beautiful but just a single loop

Reply:
cool I think it more understandable now